### ISSUE TITLE
[Git Master (d466d34)] Moving a room and clicking on the doctor causes error

### ISSUE THREAD

**Issue by mpmc** [CorsixTH/CorsixTH#362]
**Platform and build:**
Xubuntu 14.04 - git master (d466d34ef8), SDL, x64.

**Problem:**
While I was moving a room I was somehow able to click on the doctor currently leaving the room which displayed the info popup, I attempted to close the window but couldn't as I was moving the room, after I completed the room move I got an error. I've included this below..

```
Will not check for updates since this is the Trunk version.
Warning: Entity tile changed while marked as using an object
Warning: Entity tile changed while marked as using an object
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
./Lua/entities/humanoid.lua:659: Can only finish current action
stack traceback:
    [C]: in function 'assert'
    ./Lua/entities/humanoid.lua:659: in function 'finishAction'
    ./Lua/rooms/cardiogram.lua:94: in function 'after_use'
    ./Lua/humanoid_actions/multi_use_object.lua:264: in function 'timer_function'
    ./Lua/entity.lua:215: in function 'tick'
    ./Lua/world.lua:1103: in function 'onTick'
    ./Lua/app.lua:852: in function 'dispatch'
    ./Lua/app.lua:753: in function <./Lua/app.lua:748>

Warning: No event handler for timer
```

![screenshot - 051114 - 19 43 02](https://cloud.githubusercontent.com/assets/2352508/4924882/8954efde-6526-11e4-8c0f-bae9f55254f7.png)

**Comment by bobcanthelpyou** [comment]
Not sure if this is a similar issue. I moved the Pharmacy room and build a new GP's Office nearby and an error occurs:

```
Will not check for updates since this is the Trunk version.
Savegame version is 102 (Trunk), originally it was 102 (Trunk)
An error has occured!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
./Lua/humanoid_actions/multi_use_object.lua:313: attempt to index field '?' (a nil value)
stack traceback:
	./Lua/humanoid_actions/multi_use_object.lua:313: in function '?'
	./Lua/entities/humanoid.lua:575: in function 'Humanoid_startAction'
	./Lua/entities/humanoid.lua:667: in function 'finishAction'
	./Lua/humanoid_actions/walk.lua:152: in function 'timer_function'
	./Lua/entity.lua:218: in function 'tick'
	./Lua/world.lua:1121: in function 'onTick'
	./Lua/app.lua:876: in function <./Lua/app.lua:873>
	(tail call): ?
	./Lua/app.lua:777: in function <./Lua/app.lua:772>

Error in timer handler: 
./Lua/humanoid_actions/multi_use_object.lua:313: attempt to index field '?' (a nil value)
stack traceback:
	./Lua/humanoid_actions/multi_use_object.lua:313: in function '?'
	./Lua/entities/humanoid.lua:575: in function 'Humanoid_startAction'
	./Lua/entities/humanoid.lua:667: in function 'finishAction'
	./Lua/humanoid_actions/walk.lua:152: in function 'timer_function'
	./Lua/entity.lua:218: in function 'tick'
	./Lua/world.lua:1121: in function 'onTick'
	./Lua/app.lua:876: in function <./Lua/app.lua:873>
	(tail call): ?
	./Lua/app.lua:777: in function <./Lua/app.lua:772>
Warning: No event handler for timer
```
savegame before error: http://www.xup.in/dl,10243463/congame2.sav/
savegame after error: http://www.xup.in/dl,94694418/conerror.sav/ (the room with the cast remover seems problematic)

**Comment by J-Shep** [comment]
I've not reproduced this error yet but I have a new idea for its cause: in the screenshot provided by @mpmc a doctor with black skin can be seen walking away from the cardiogram room where the "staff cardiogram user not idle after_use" error took place. So he may have been the doctor working in this room when the patient started using the cardiogram and then when he was standing idle using it calls_dispatcher:verifyStaffForRoom() would have seen he was idle and may have concluded that he was available to send to another room which required a doctor.

**Comment by J-Shep** [comment]
The calls dispatcher does conclude that the staff member is available when they are actually using the cardiogram / general diagnosis trolley but this just ends the use of the room's object early for both the patient and doctor so no error occurs. But this issue's error occurred in 0.40 so perhaps changes this release have prevented this error from occurring.

If the calls dispatcher was responsible it would explain why after editing rooms @mpmc  and @MarkL1961 experienced this error in their hospital's cardiogram/general diagnosis room (the only multi_use_object rooms which have the patient running this action while the staff member is idle) because when these room's were finished patients' room built callbacks could have sent them to these rooms which had no staff making the calls dispatcher request staff.

**Comment by J-Shep** [comment]
>> But this issue's error occurred in 0.40 so perhaps changes since this release have prevented this error from occurring.

I've tried 0.40 and my idea doesn't cause this error.

**Comment by J-Shep** [comment]
If anyone experiences this error again and can provide a saved game from just before it happened or afterwards then please upload your saved game and post a link to it here.

I've not been able to reproduce this error and I don't have any more ideas for what its cause could be so I can't make any progress with investigating it.

**Comment by bobcanthelpyou** [comment]
@J-Shep there are some issues available with a similar 'Can only finish current action' stack traceback
#858 #650 #500 maybe the savegames of these issues could help you.

**Comment by J-Shep** [comment]
Thanks for your helpful suggestion @bobcanthelpyou I will try using their saved games.

**Comment by mugmuggy** [comment]
Fixed with #1567
